sagemaker-fundamentals/create-endpoint/container/src/predictor.py

```diff
@@ -0,0 +1,42 @@
+# This is the file that implements a flask server to do inferences. It's the file that you will modify to adapt to your own inference logic.
+
+import flask
+from flask import request
+
+# The flask app for serving predictions
+app = flask.Flask(__name__)
+
+@app.route('/ping', methods=['GET'])
+def ping():
+    """Determine if the container is working and healthy. 
+    In this sample container, we declare
+    it healthy if we can load the model successfully."""
+
+    status = 200
+    return flask.Response(response='\n', 
+        status=status, mimetype='application/json')
+
+@app.route('/invocations', methods=['POST'])
+def inference():
```

Comment:
Does this method actually do inference? It seems like it only prints. If so, then the method name should be something slightly different.

Reply:
`inference` is what will be called when you invoke the endpoint with SageMaker runtime client. In real life an ML practitioner should write the inference logic under `def inference`. but here I just want to demonstrate that `def inference` is indeeded being called when we invoke the endpoint with sagemaker runtime. It prints out the request, indicating that it received the request